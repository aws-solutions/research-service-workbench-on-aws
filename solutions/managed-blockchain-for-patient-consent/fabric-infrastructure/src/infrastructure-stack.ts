/*
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *  SPDX-License-Identifier: Apache-2.0
 */
import { Stack, StackProps } from 'aws-cdk-lib';
import * as cdk from 'aws-cdk-lib';
import { NagSuppressions } from 'cdk-nag';
import { Construct } from 'constructs';
import { isEmpty } from 'lodash';

import * as hyperledger from './components';

export class InfrastructureStack extends Stack {
  public constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);
    const networkId = this.node.tryGetContext('networkId');
    const invitationId = this.node.tryGetContext('invitationId');
    const additionalMembers = this.node.tryGetContext('additionalMembers');
    const availabilityZone = isEmpty(this.node.tryGetContext('availabilityZone'))
      ? Stack.of(this).availabilityZones[0]
      : this.node.tryGetContext('availabilityZone');
    const accountId = cdk.Stack.of(this).account;
    const network = new hyperledger.HyperledgerFabricNetwork(this, 'V2TestNetwork', {
      networkName: 'TestNetwork',
      memberName: `TestMember-${accountId}`,
      networkEdition: hyperledger.NetworkEdition.STARTER,
      networkId,
      invitationId,
      additionalMembers,
      nodes: [
        {
          availabilityZone,
          instanceType: hyperledger.InstanceType.BURSTABLE3_MEDIUM
        }
      ]
    });

    /* eslint-disable no-new */
    new cdk.CfnOutput(this, 'NetworkId', {
      description: 'Managed Blockchain network identifier',
      value: network.networkId
    });

    new cdk.CfnOutput(this, 'MemberId', {
      description: 'Managed Blockchain member identifier',
      value: network.memberId
    });

    new cdk.CfnOutput(this, 'VpcEndpointServiceName', {
      description: 'Managed Blockchain network VPC endpoint service name',
      value: network.vpcEndpointServiceName
    });

    new cdk.CfnOutput(this, 'OrdererEndpoint', {
      description: 'Managed Blockchain network ordering service endpoint',
      value: network.ordererEndpoint
    });

    new cdk.CfnOutput(this, 'CaEndpoint', {
      description: 'Managed Blockchain member CA endpoint',
      value: network.caEndpoint
    });

    new cdk.CfnOutput(this, 'AdminPasswordArn', {
      description: 'Secret ARN for the Hyperledger Fabric admin password',
      value: network.adminPasswordSecret.secretFullArn ?? network.adminPasswordSecret.secretArn
    });

    new cdk.CfnOutput(this, 'AdminPrivateKeyArn', {
      description: 'Secret ARN for Hyperledger Fabric admin private key',
      value: network.adminPrivateKeySecret.secretFullArn ?? network.adminPrivateKeySecret.secretArn
    });

    new cdk.CfnOutput(this, 'AdminSignedCertArn', {
      description: 'Secret ARN for Hyperledger Fabric admin signed certificate',
      value: network.adminSignedCertSecret.secretFullArn ?? network.adminSignedCertSecret.secretArn
    });

    new cdk.CfnOutput(this, 'NodeIds', {
      description: 'Comma-separated list of Managed Blockchain node identifiers',
      value: network.nodes.map((n) => n.nodeId).join(',')
    });

    new cdk.CfnOutput(this, 'NodeEndpoints', {
      description: 'Comma-separated list of Managed Blockchain node endpoints',
      value: network.nodes.map((n) => n.endpoint).join(',')
    });

    new cdk.CfnOutput(this, 'NodeEventEndpoints', {
      description: 'Comma-separated list of Managed Blockchain node event endpoints',
      value: network.nodes.map((n) => n.eventEndpoint).join(',')
    });

    /**
     * cdk-nag suppression
     */
    NagSuppressions.addStackSuppressions(this, [
      {
        id: 'AwsSolutions-IAM4',
        reason:
          'The CDK custom resource framework uses a managed policy for its Lambda, and the name for the Lambda is randomly generated'
      },
      {
        id: 'AwsSolutions-IAM5',
        reason:
          'The CDK custom resource framework uses wildcard permission for its Lambda, and the name for the Lambda is randomly generated'
      },
      {
        id: 'AwsSolutions-L1',
        reason:
          'The CDK custom resource framework uses NodeJS 12 and NodeJS 14 for onEvent trigger, and the name for these resources are randomly generated'
      }
    ]);

    NagSuppressions.addResourceSuppressionsByPath(
      this,
      '/HyperledgerTestStack/V2TestNetwork/AdminPassword/Resource',
      [
        {
          id: 'AwsSolutions-SMG4',
          reason: 'Secrets created for Managed Blockchain users do not support auto-rotation'
        }
      ]
    );

    NagSuppressions.addResourceSuppressionsByPath(
      this,
      '/HyperledgerTestStack/V2TestNetwork/AdminPrivateKey/Resource',
      [
        {
          id: 'AwsSolutions-SMG4',
          reason: 'Secrets created for Managed Blockchain users do not support auto-rotation'
        }
      ]
    );

    NagSuppressions.addResourceSuppressionsByPath(
      this,
      '/HyperledgerTestStack/V2TestNetwork/AdminSignedCert/Resource',
      [
        {
          id: 'AwsSolutions-SMG4',
          reason: 'Secrets created for Managed Blockchain users do not support auto-rotation'
        }
      ]
    );
  }
}
