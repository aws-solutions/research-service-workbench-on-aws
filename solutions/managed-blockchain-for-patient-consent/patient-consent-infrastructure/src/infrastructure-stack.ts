/* eslint-disable no-new */
import { join } from 'path';
import {
  Stack,
  StackProps,
  CfnOutput,
  RemovalPolicy,
  aws_apigateway,
  aws_kms,
  Duration,
  Fn
} from 'aws-cdk-lib';
import {
  ApiKeySourceType,
  AuthorizationType,
  CognitoUserPoolsAuthorizer,
  RestApi,
  EndpointType,
  LambdaIntegration,
  MethodLoggingLevel,
  AccessLogFormat,
  LogGroupLogDestination,
  LambdaRestApi,
  RequestAuthorizer,
  RequestValidator,
  AccessLogField
} from 'aws-cdk-lib/aws-apigateway';

import {
  AccountRootPrincipal,
  Effect,
  PolicyDocument,
  PolicyStatement,
  ServicePrincipal
} from 'aws-cdk-lib/aws-iam';
import { Code, Function, Runtime } from 'aws-cdk-lib/aws-lambda';
import { LogGroup } from 'aws-cdk-lib/aws-logs';
import { NagSuppressions } from 'cdk-nag';
import { Construct } from 'constructs';

export class InfrastructureStack extends Stack {
  public constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);

    const lambdaService = new Function(this, 'LambdaService', {
      runtime: Runtime.NODEJS_14_X,
      handler: 'buildLambda.handler',
      code: Code.fromAsset(join(__dirname, '../build')),
      functionName: 'LambdaService'
    });

    const kmsKeyPolicyDocument = new PolicyDocument({
      statements: [
        new PolicyStatement({
          actions: ['kms:*'],
          effect: Effect.ALLOW,
          principals: [new AccountRootPrincipal()],
          resources: ['*']
        }),
        new PolicyStatement({
          actions: [
            'kms:Encrypt*',
            'kms:Decrypt*',
            'kms:ReEncrypt*',
            'kms:GenerateDataKey*',
            'kms:Describe*'
          ],
          effect: Effect.ALLOW,
          principals: [new ServicePrincipal(Fn.sub('logs.${AWS::Region}.amazonaws.com'))],
          resources: ['*']
        })
      ]
    });

    const kmsKey = new aws_kms.Key(this, 'KMSKey', {
      enableKeyRotation: true,
      removalPolicy: RemovalPolicy.DESTROY,
      pendingWindow: Duration.days(7),
      policy: kmsKeyPolicyDocument
    });

    const logGroup = new LogGroup(this, 'log', {
      logGroupName: 'restApiLogGroup',
      encryptionKey: kmsKey
    });

    const restApi = new LambdaRestApi(this, 'RestApi', {
      handler: lambdaService,
      proxy: false,
      cloudWatchRole: false,
      deployOptions: {
        accessLogDestination: new LogGroupLogDestination(logGroup),
        accessLogFormat: AccessLogFormat.custom(
          `${AccessLogField.contextRequestId()} ${AccessLogField.contextErrorMessage()} ${AccessLogField.contextErrorMessageString()}`
        )
      }
    });

    const requestValidator = new RequestValidator(this, 'MyRequestValidator', {
      restApi: restApi,
      requestValidatorName: 'requestValidatorName',
      validateRequestBody: false,
      validateRequestParameters: false
    });

    const patient = restApi.root.addResource('patient');
    patient.addResource('signup').addMethod('POST', new LambdaIntegration(lambdaService), {
      requestValidator: requestValidator,
      authorizationType: AuthorizationType.IAM,
      apiKeyRequired: false
    });

    const organization = restApi.root.addResource('organization');
    organization.addResource('set-org').addMethod('POST', new LambdaIntegration(lambdaService), {
      requestValidator: requestValidator,
      authorizationType: AuthorizationType.IAM,
      apiKeyRequired: false
    });
    organization.addResource('get-org').addMethod('POST', new LambdaIntegration(lambdaService), {
      requestValidator: requestValidator,
      authorizationType: AuthorizationType.IAM,
      apiKeyRequired: false
    });

    const consent = restApi.root.addResource('consent');
    consent.addResource('set-consent').addMethod('POST', new LambdaIntegration(lambdaService), {
      requestValidator: requestValidator,
      authorizationType: AuthorizationType.IAM,
      apiKeyRequired: false
    });
    consent.addResource('get-consent').addMethod('POST', new LambdaIntegration(lambdaService), {
      requestValidator: requestValidator,
      authorizationType: AuthorizationType.IAM,
      apiKeyRequired: false
    });
    consent.addResource('revoke').addMethod('POST', new LambdaIntegration(lambdaService), {
      requestValidator: requestValidator,
      authorizationType: AuthorizationType.IAM,
      apiKeyRequired: false
    });

    new CfnOutput(this, 'HttpEndpoint', {
      value: restApi.url,
      exportName: 'HttpApiEndPoint'
    });

    /**
     * cdk-nag suppressions
     */
    NagSuppressions.addResourceSuppressionsByPath(
      this,
      '/InfrastructureStack/LambdaService/ServiceRole/Resource',
      [{ id: 'AwsSolutions-IAM4', reason: "I'm ok using managed policies for this example" }]
    );
    NagSuppressions.addResourceSuppressionsByPath(
      this,
      '/InfrastructureStack/RestApi/Default/organization/set-org/POST/Resource',
      [{ id: 'AwsSolutions-COG4', reason: 'We will use IAM and API key' }]
    );
    NagSuppressions.addResourceSuppressionsByPath(
      this,
      '/InfrastructureStack/RestApi/Default/organization/get-org/POST/Resource',
      [{ id: 'AwsSolutions-COG4', reason: 'We will use IAM and API key' }]
    );
    NagSuppressions.addResourceSuppressionsByPath(
      this,
      '/InfrastructureStack/RestApi/Default/consent/revoke/POST/Resource',
      [{ id: 'AwsSolutions-COG4', reason: 'We will use IAM and API key' }]
    );
    NagSuppressions.addResourceSuppressionsByPath(
      this,
      '/InfrastructureStack/RestApi/Default/consent/set-consent/POST/Resource',
      [{ id: 'AwsSolutions-COG4', reason: 'We will use IAM and API key' }]
    );
    NagSuppressions.addResourceSuppressionsByPath(
      this,
      '/InfrastructureStack/RestApi/Default/consent/get-consent/POST/Resource',
      [{ id: 'AwsSolutions-COG4', reason: 'We will use IAM and API key' }]
    );
    NagSuppressions.addResourceSuppressionsByPath(
      this,
      '/InfrastructureStack/RestApi/Default/patient/signup/POST/Resource',
      [{ id: 'AwsSolutions-COG4', reason: 'We will use IAM and API key' }]
    );
    NagSuppressions.addResourceSuppressionsByPath(
      this,
      '/InfrastructureStack/RestApi/DeploymentStage.prod/Resource',
      [
        {
          id: 'AwsSolutions-APIG3',
          reason: 'Access is configured by IAM role and API key'
        },
        {
          id: 'AwsSolutions-APIG6',
          reason: 'We will not be using CloudWatch not needed at this time'
        }
      ]
    );
    NagSuppressions.addResourceSuppressionsByPath(this, '/InfrastructureStack/LambdaService/Resource', [
      {
        id: 'AwsSolutions-L1',
        reason: 'We are using Node14 for Lambda functions'
      }
    ]);
  }
}
