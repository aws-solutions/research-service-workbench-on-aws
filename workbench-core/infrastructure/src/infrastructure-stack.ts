import { Stack, StackProps, CfnOutput } from 'aws-cdk-lib';
import { Code, Function, Runtime } from 'aws-cdk-lib/aws-lambda';
import { Construct } from 'constructs';
import { HttpApi } from '@aws-cdk/aws-apigatewayv2-alpha';
import { HttpLambdaIntegration } from '@aws-cdk/aws-apigatewayv2-integrations-alpha';
// import { LogGroup, RetentionDays } from 'aws-cdk-lib/aws-logs';
// import { Effect, PolicyStatement, Role, ServicePrincipal } from 'aws-cdk-lib/aws-iam';

export class InfrastructureStack extends Stack {
  // eslint-disable-next-line @typescript-eslint/explicit-member-accessibility
  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);

    // const logGroup = new LogGroup(this, 'LogGroup', {
    //   retention: RetentionDays.ONE_WEEK,
    //   logGroupName: 'HttpApiLogGroup'
    // });

    // const lambdaRole = new Role(this, 'LambdaRole', {
    //   assumedBy: new ServicePrincipal('lambda.amazonaws.com')
    // });

    // lambdaRole.addToPolicy(
    //   new PolicyStatement({
    //     effect: Effect.ALLOW,
    //     actions: ['logs:CreateLogGroup', 'logs:CreateLogStream', 'logs:PutLogEvents'],
    //     resources: [logGroup.logGroupArn]
    //   })
    // );

    const lambdaService = new Function(this, 'LambdaService', {
      runtime: Runtime.NODEJS_14_X,
      handler: 'index.handler',
      code: Code.fromAsset('../infrastructure/lambda-build/archive.zip'),
      functionName: 'LambdaService'
      // role: lambdaRole
    });

    const httpLambdaIntegration = new HttpLambdaIntegration('HttpLambda', lambdaService);

    const httpApi = new HttpApi(this, 'HttpApi', {
      defaultIntegration: httpLambdaIntegration,
      apiName: 'HttpApiService'
    });

    // const httpApiStage = new HttpStage(this, 'HttpStage', {
    //   httpApi,
    //   autoDeploy: true,
    //   stageName: "$default"
    // })

    // const defaultStage = httpApi.defaultStage?.node.defaultChild as CfnStage;

    // defaultStage.accessLogSettings = {
    //   destinationArn: logGroup.logGroupArn,
    //   format: `$context.identity.sourceIp - - [$context.requestTime] "$context.httpMethod $context.routeKey $context.protocol" $context.status $context.responseLength $context.requestId`
    // };

    // eslint-disable-next-line no-new
    new CfnOutput(this, 'HttpEndpoint', {
      value: httpApi.apiEndpoint,
      exportName: 'HttpApiEndPoint'
    });
  }
}
