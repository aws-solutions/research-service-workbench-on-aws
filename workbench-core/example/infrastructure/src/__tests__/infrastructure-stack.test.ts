import { App } from 'aws-cdk-lib';
import { expect as expectCDK, haveResourceLike } from '@aws-cdk/assert';
import { InfrastructureStack } from '../infrastructure-stack';
import { Template } from 'aws-cdk-lib/assertions';

describe('CDKStack', () => {
  test('Infrastructure Test', () => {
    const app = new App();
    const infraStack = new InfrastructureStack(app, 'InfrastructureStack');
    expectCDK(infraStack).to(
      haveResourceLike('AWS::Lambda::Function', {
        FunctionName: 'LambdaService'
      })
    );
    expectCDK(infraStack).to(
      haveResourceLike('AWS::ApiGatewayV2::Integration', {
        IntegrationType: 'AWS_PROXY'
      })
    );
    expectCDK(infraStack).to(
      haveResourceLike('AWS::ApiGatewayV2::Api', {
        Name: 'HttpApiService'
      })
    );
    const infraStackTemplate = Template.fromStack(infraStack);
    expect(infraStackTemplate).toMatchSnapshot();
  });
});
