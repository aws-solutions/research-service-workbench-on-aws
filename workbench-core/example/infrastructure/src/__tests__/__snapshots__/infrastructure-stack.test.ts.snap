// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CDKStack Infrastructure Test 1`] = `
Object {
  "Outputs": Object {
    "HttpEndpoint": Object {
      "Export": Object {
        "Name": "HttpApiEndPoint",
      },
      "Value": Object {
        "Fn::GetAtt": Array [
          "HttpApiF5A9A8A7",
          "ApiEndpoint",
        ],
      },
    },
    "userPoolClientId": Object {
      "Value": Object {
        "Ref": "ExampleUserPoolClientA3BAC0B9",
      },
    },
    "userPoolId": Object {
      "Value": Object {
        "Ref": "ExampleUserPoolB793288C",
      },
    },
  },
  "Resources": Object {
    "ExampleUserPoolB793288C": Object {
      "DeletionPolicy": "Delete",
      "Metadata": Object {
        "cdk_nag": Object {
          "rules_to_suppress": Array [
            Object {
              "id": "AwsSolutions-COG2",
              "reason": "I'm ok with not using MFA Authentication for this example",
            },
            Object {
              "id": "AwsSolutions-COG3",
              "reason": "UserPool CDK construct does not have option to set AdvancedSecurityMode",
            },
          ],
        },
      },
      "Properties": Object {
        "AccountRecoverySetting": Object {
          "RecoveryMechanisms": Array [
            Object {
              "Name": "verified_email",
              "Priority": 1,
            },
          ],
        },
        "AdminCreateUserConfig": Object {
          "AllowAdminCreateUserOnly": false,
        },
        "AliasAttributes": Array [
          "email",
        ],
        "AutoVerifiedAttributes": Array [
          "email",
        ],
        "EmailVerificationMessage": "The verification code to your new account is {####}",
        "EmailVerificationSubject": "Verify your new account",
        "Policies": Object {
          "PasswordPolicy": Object {
            "MinimumLength": 8,
            "RequireLowercase": true,
            "RequireNumbers": true,
            "RequireSymbols": true,
            "RequireUppercase": true,
          },
        },
        "SmsVerificationMessage": "The verification code to your new account is {####}",
        "UserPoolName": "example-user-pool",
        "VerificationMessageTemplate": Object {
          "DefaultEmailOption": "CONFIRM_WITH_CODE",
          "EmailMessage": "The verification code to your new account is {####}",
          "EmailSubject": "Verify your new account",
          "SmsMessage": "The verification code to your new account is {####}",
        },
      },
      "Type": "AWS::Cognito::UserPool",
      "UpdateReplacePolicy": "Delete",
    },
    "ExampleUserPoolClientA3BAC0B9": Object {
      "Properties": Object {
        "AllowedOAuthFlows": Array [
          "implicit",
          "code",
        ],
        "AllowedOAuthFlowsUserPoolClient": true,
        "AllowedOAuthScopes": Array [
          "profile",
          "phone",
          "email",
          "openid",
          "aws.cognito.signin.user.admin",
        ],
        "CallbackURLs": Array [
          "https://example.com",
        ],
        "ExplicitAuthFlows": Array [
          "ALLOW_USER_PASSWORD_AUTH",
          "ALLOW_ADMIN_USER_PASSWORD_AUTH",
          "ALLOW_CUSTOM_AUTH",
          "ALLOW_USER_SRP_AUTH",
          "ALLOW_REFRESH_TOKEN_AUTH",
        ],
        "SupportedIdentityProviders": Array [
          "COGNITO",
        ],
        "UserPoolId": Object {
          "Ref": "ExampleUserPoolB793288C",
        },
      },
      "Type": "AWS::Cognito::UserPoolClient",
    },
    "HttpApiDefaultRoute146696CE": Object {
      "Properties": Object {
        "ApiId": Object {
          "Ref": "HttpApiF5A9A8A7",
        },
        "AuthorizationType": "JWT",
        "AuthorizerId": Object {
          "Ref": "HttpApiuserpoolauthorizerFEBA2CB6",
        },
        "RouteKey": "$default",
        "Target": Object {
          "Fn::Join": Array [
            "",
            Array [
              "integrations/",
              Object {
                "Ref": "HttpApiDefaultRouteHttpLambda91177580",
              },
            ],
          ],
        },
      },
      "Type": "AWS::ApiGatewayV2::Route",
    },
    "HttpApiDefaultRouteHttpLambda91177580": Object {
      "Properties": Object {
        "ApiId": Object {
          "Ref": "HttpApiF5A9A8A7",
        },
        "IntegrationType": "AWS_PROXY",
        "IntegrationUri": Object {
          "Fn::GetAtt": Array [
            "LambdaService68B9B9DF",
            "Arn",
          ],
        },
        "PayloadFormatVersion": "2.0",
      },
      "Type": "AWS::ApiGatewayV2::Integration",
    },
    "HttpApiDefaultRouteHttpLambdaPermissionDE2D6043": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "LambdaService68B9B9DF",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              Object {
                "Ref": "AWS::Region",
              },
              ":",
              Object {
                "Ref": "AWS::AccountId",
              },
              ":",
              Object {
                "Ref": "HttpApiF5A9A8A7",
              },
              "/*/*",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "HttpApiDefaultStage3EEB07D6": Object {
      "Properties": Object {
        "AccessLogSettings": Object {
          "DestinationArn": Object {
            "Fn::GetAtt": Array [
              "log751C81CB",
              "Arn",
            ],
          },
          "Format": "$context.identity.sourceIp - - [$context.requestTime] \\"$context.httpMethod $context.routeKey $context.protocol\\" $context.status $context.responseLength $context.requestId",
        },
        "ApiId": Object {
          "Ref": "HttpApiF5A9A8A7",
        },
        "AutoDeploy": true,
        "StageName": "$default",
      },
      "Type": "AWS::ApiGatewayV2::Stage",
    },
    "HttpApiF5A9A8A7": Object {
      "Properties": Object {
        "Name": "HttpApiService",
        "ProtocolType": "HTTP",
      },
      "Type": "AWS::ApiGatewayV2::Api",
    },
    "HttpApiuserpoolauthorizerFEBA2CB6": Object {
      "Properties": Object {
        "ApiId": Object {
          "Ref": "HttpApiF5A9A8A7",
        },
        "AuthorizerType": "JWT",
        "IdentitySource": Array [
          "$request.header.Authorization",
        ],
        "JwtConfiguration": Object {
          "Audience": Array [
            Object {
              "Ref": "ExampleUserPoolClientA3BAC0B9",
            },
          ],
          "Issuer": Object {
            "Fn::Join": Array [
              "",
              Array [
                "https://cognito-idp.",
                Object {
                  "Ref": "AWS::Region",
                },
                ".amazonaws.com/",
                Object {
                  "Ref": "ExampleUserPoolB793288C",
                },
              ],
            ],
          },
        },
        "Name": "user-pool-authorizer",
      },
      "Type": "AWS::ApiGatewayV2::Authorizer",
    },
    "KMSKeyBD866E3F": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "EnableKeyRotation": true,
        "KeyPolicy": Object {
          "Statement": Array [
            Object {
              "Action": "kms:*",
              "Effect": "Allow",
              "Principal": Object {
                "AWS": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::",
                      Object {
                        "Ref": "AWS::AccountId",
                      },
                      ":root",
                    ],
                  ],
                },
              },
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::KMS::Key",
      "UpdateReplacePolicy": "Retain",
    },
    "LambdaService68B9B9DF": Object {
      "DependsOn": Array [
        "LambdaServiceServiceRole71319041",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "0cbba7608888c3948fdec181048880eebc1eca58d3e2f70826988b9abfb1076a.zip",
        },
        "FunctionName": "LambdaService",
        "Handler": "buildLambda.handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "LambdaServiceServiceRole71319041",
            "Arn",
          ],
        },
        "Runtime": "nodejs14.x",
      },
      "Type": "AWS::Lambda::Function",
    },
    "LambdaServiceServiceRole71319041": Object {
      "Metadata": Object {
        "cdk_nag": Object {
          "rules_to_suppress": Array [
            Object {
              "id": "AwsSolutions-IAM4",
              "reason": "I'm ok using managed policies for this example",
            },
          ],
        },
      },
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "log751C81CB": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "KmsKeyId": Object {
          "Fn::GetAtt": Array [
            "KMSKeyBD866E3F",
            "Arn",
          ],
        },
        "LogGroupName": "httpApiLogGroup",
        "RetentionInDays": 731,
      },
      "Type": "AWS::Logs::LogGroup",
      "UpdateReplacePolicy": "Retain",
    },
  },
}
`;
